crates/jit/src/mmap_vec.rs

```diff
@@ -73,6 +76,26 @@ impl MmapVec {
         }
     }
 
+    /// Creates a new `MmapVec` which is the `path` specified mmap'd into
+    /// memory.
+    ///
+    /// This function will attempt to open the file located at `path` and will
+    /// then use that file to learn about its size and map the full contents
+    /// into memory. This will return an error if the file doesn't exist or if
+    /// it's too large to be fully mapped into memory.
+    pub fn from_file(path: &Path) -> Result<MmapVec> {
+        let file =
+            File::open(path).with_context(|| format!("failed to open: {}", path.display()))?;
+        let len = file
+            .metadata()
+            .context("failed to get file metadata")?
```

Comment:
I opted to add this context a layer above to avoid having to include it in all the error messages here (if you're ok with that since at least for now it's only called in one location)